Monika After Story/game/zz_windowutils.rpy

```diff
@@ -0,0 +1,746 @@
+            active_winname_prop = active_winobj.get_full_property(NET_WM_NAME, 0)
+
+            if active_winname_prop is not None:
+                active_winname = unicode(active_winname_prop.value)
+                return (
+                    active_winname.replace("\n", "")
+                    if friendly
+                    else active_winname.lower().replace(" ", "").replace("\n", "")
+                )
+
+            else:
+                return ""
+
+        except BadWindow:
+            return ""
+
+    def _getActiveWindow_OSX(friendly):
+        """
+        Gets the active window on macOS
+
+        IN:
+            friendly - Whether or not we want the output in a usable state
+
+        NOTE: This currently just returns an empty string, this is because we do not have active window detection
+        for MacOS
+        """
+        return ""
+
+    #Notif show internals
+    def _tryShowNotification_Windows(title, body):
+        """
+        Tries to push a notification to the notification center on Windows.
+        If it can't it should fail silently to the user.
+
+        IN:
+            title - notification title
+            body - notification body
+
+        OUT:
+            bool. True if the notification was successfully sent, False otherwise
+        """
+        # The Windows way, notif_success is adjusted if need be
+        notif_success = __tip.showWindow(title, body)
+
+        #We need the IDs of the notifs to delete them from the tray
+        store.destroy_list.append(__tip.hwnd)
+        return notif_success
+
+    def _tryShowNotification_Linux(title, body):
+        """
+        Tries to push a notification to the notification center on Linux.
+        If it can't it should fail silently to the user.
+
+        IN:
+            title - notification title
+            body - notification body
+
+        OUT:
+            bool - True, representing the notification's success
+        """
+        # Single quotes have to be escaped.
+        # Since single quoting in POSIX shell doesn't allow escaping,
+        # we have to close the quotation, insert a literal single quote and reopen the quotation.
+        body  = body.replace("'", "'\\''")
+        title = title.replace("'", "'\\''") # better safe than sorry
+        os.system("notify-send '{0}' '{1}' -a 'Monika' -u low".format(title, body))
+        return True
+
+    def _tryShowNotification_OSX(title, body):
+        """
+        Tries to push a notification to the notification center on macOS.
+        If it can't it should fail silently to the user.
+
+        IN:
+            title - notification title
+            body - notification body
+
+        OUT:
+            bool - True, representing the notification's success
+        """
+        os.system('osascript -e \'display notification "{0}" with title "{1}"\''.format(renpy.substitute(body, title)))
```

Comment:
This one makes notifications crash as renpy.subsitute only returns a single string and we need two to properly format this one.
Something like this should do if you need to do the substitution on the title too.
```
os.system('osascript -e \'display notification "{0}" with title "{1}"\''.format(renpy.substitute(body), renpy.substitute(title)))
```